### ISSUE TITLE
bidirect_index pattern

### ISSUE THREAD

**Issue by yegor256** [yegor256/aibolit#139]
Here it is:

```java
class Foo {
  void bar() {
    int i = 0;
    i++;
    i--;
  }
}
```

If the same numeric variable is incremented and decremented within the same method, it's a pattern. A numeric variable should either be always growing or decreasing. Bi-directional index is confusing.

**Comment by 0crat** [comment]
@acheshkov[/z](https://www.0crat.com/u/acheshkov) please, pay attention to this issue

**Comment by acheshkov** [comment]
@0crat in

**Comment by 0crat** [comment]
The job [#139](https://github.com/yegor256/aibolit/issues/139) assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/373914c2-38d8-436c-ba42-23b6a8891a9e); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by paulodamaso** [comment]
@0crat wait for #242 merge

**Comment by paulodamaso** [comment]
@acheshkov @yegor256 Merged, please close

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#282](https://github.com/yegor256/aibolit/issues/282) is still not solved.

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#486](https://github.com/cqfn/aibolit/issues/486) is still not solved.

**Comment by 0crat** [comment]
I'm not managing `cqfn/aibolit` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by 0crat** [comment]
I'm not managing `cqfn/aibolit` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#742](https://github.com/cqfn/aibolit/issues/742) is still not solved; solved: [#486](https://github.com/cqfn/aibolit/issues/486).

**Comment by 0pdd** [comment]
@yegor256 all 2 puzzles are solved here: [#486](https://github.com/cqfn/aibolit/issues/486), [#742](https://github.com/cqfn/aibolit/issues/742).
